### PR TITLE
🐛 [BUG:] Fixed warnings and a11y issues

### PR DESCRIPTION
**Pull request by amaan-ahmad**
- added alt-text to images (wherever necessary)
- fixed target='_blank' vulnerability, 
- fixed react useEffect() dependencies missing warnings and tested with the socket server✔
- other minor fixes.

PS: there might be some extra whitespace added in lines because of a different document formatter. (this won't affect the working of code)

### CODE REVIEW

**Comment by amaan-ahmad** [comment]
Hey @kothariji kindly review this PR associated with issue #78

**Comment by Vishv07** [comment]
LGTM🔥

**Comment by kothariji** [comment]
@amaan-ahmad 
Build is failing, here is the error message. Please rectify it.
image

File: frontend/syntaxmeets/src/components/SyntaxChat/SyntaxChat.js [hunk 1]
```diff
@@ -63,6 +65,7 @@ const SyntaxChat = (props) => {
     });
   }, []);
```
Comment (Vishv07) [inline frontend/syntaxmeets/src/components/SyntaxChat/SyntaxChat.js:66]: pass a message or data as an optional second argument to useEffect.

**Comment by amaan-ahmad** [comment]
@kothariji  that was caused due to conflict with another commit feature of 'someone is typing...', 
I resolved it now.

image

**Comment by amaan-ahmad** [comment]
Hey @kothariji , do let me know if there are further changes or we are good to merge ✌🙌

**Comment by kothariji** [comment]
> Hey @kothariji , do let me know if there are further changes or we are good to merge ✌🙌

Yes we are reviewing it, and once done, we'll let you know. 🙌

**Review by kothariji — CHANGES_REQUESTED** [review]
Actually, there are so many issues I am facing in this PR.

Here is the preview link of PR - https://syntaxmeets-git-removewarnings-kothariji.vercel.app/

Problems - 1
Whenever 2nd user joins into the room, the entire code is replaced with the default code. Which is not the kind of behavior it should show.

Problem - 2 
2nd user is not able to type properly in CodeEditor. The cursor is changing positions,.

So please look over these issue. The are mostly because of useEffect. 

For eg. 
you have replaced this -> `},[]);`
  with `}, [props.socket]);`    in syntax editor line 101. But this is wrong. Actually we need to call that useEffect only one, therefore we add an empty array. 

So, please consider all these points.

**Comment by amaan-ahmad** [comment]
hey @kothariji 

Yes, I found this error hence adding props and socket to the dependency array wasn't a fruitful decision at my end, hence I am reverting it. However other securities and a11y warnings have been fixed now in the latest commit. Kindly check, waiting for feedback.

And thanks for patiently providing a review. 😊🙌

Best regards.

File: frontend/syntaxmeets/src/components/SyntaxEditor/SyntaxEditor.js [hunk 2]
```diff
@@ -85,20 +85,18 @@ const SyntaxEditor = (props) => {
   const [codeError, setCodeError] = useState("");
   // This will resend a message to update the code of the newly joined user
   useEffect(() => {
-
     if (props.previousUser.id === props.id) {
-
       props.socket.emit("message", value);
     }
-  }, [props.previousUser]);
+  }, []);
```
Comment (kothariji) [inline frontend/syntaxmeets/src/components/SyntaxEditor/SyntaxEditor.js:92]: There is a small mistake, you removed props.previousUser, due to which code is not updated for a new user when he joins, Please rectify it 🙏

**Review by kothariji — CHANGES_REQUESTED** [review]
Please do the following changes

File: frontend/syntaxmeets/src/components/SyntaxEditor/SyntaxEditor.js (hunk 2 above)
Comment (amaan-ahmad) [reply frontend/syntaxmeets/src/components/SyntaxEditor/SyntaxEditor.js]: Oh yes, thanks for letting me know. Fixed it. 🔨

**Review by kothariji — APPROVED** [review]
LGTM 🚀

**Pull request merged**